src/client/meshgen/collector.h

```diff
@@ -40,9 +41,10 @@ struct MeshCollector
 	// bounding sphere radius and center
 	f32 m_bounding_radius_sq = 0.0f;
 	v3f m_center_pos;
+	v3f offset;
 
 	// center_pos: pos to use for bounding-sphere, in BS-space
-	MeshCollector(const v3f center_pos) : m_center_pos(center_pos) {}
+	MeshCollector(const v3f center_pos, v3f _offset) : m_center_pos(center_pos), offset(_offset) {}
```

Comment:
```suggestion
	// offset: offset added to vertices
	MeshCollector(const v3f center_pos, v3f offset) : m_center_pos(center_pos), offset(offset) {}
```

this would also eliminate the change in src/client/wieldmesh.cpp